packages/lesswrong/server/resolvers/notificationResolvers.ts

```diff
@@ -141,3 +142,21 @@ createPaginatedResolver({
     });
   },
 });
+
+addGraphQLResolvers({
+  Mutation: {
+    MarkAllNotificationsAsRead: async (
+      _root: void,
+      _args: {},
+      {currentUser, repos}: ResolverContext,
+    ) => {
+      if (!currentUser) {
+        throw new Error("Unauthorized");
+      }
+      await repos.notifications.markAllAsRead(currentUser._id);
```

Comment:
I think this can be a `Notifications.rawUpdateMany({ userId: currentUser._id, type: { $ne: 'newMessage' } }, { viewed: true })`?